src/main/java/org/truffleruby/core/string/StringNodes.java

```diff
@@ -5105,16 +5106,42 @@ protected boolean indexAtEitherBounds(RubyString string, int index) {
 
     }
 
+    @ImportStatic(RopeGuards.class)
     @Primitive(name = "string_to_inum", lowerFixnum = 1)
-    public abstract static class StringToInumPrimitiveNode extends PrimitiveArrayArgumentsNode {
+    @NodeChild(value = "string", type = RubyNode.class)
+    @NodeChild(value = "fixBase", type = RubyNode.class)
+    @NodeChild(value = "strict", type = RubyNode.class)
+    @NodeChild(value = "raiseOnError", type = RubyNode.class)
+    public abstract static class StringToInumPrimitiveNode extends PrimitiveNode {
+
+        @Child FixnumOrBignumNode fixnumOrBignumNode;
+        @Child RopeNodes.BytesNode bytesNode;
+
+        @CreateCast("string")
+        protected RubyNode coerceStringToRope(RubyNode string) {
+            return ToRopeNodeGen.create(string);
+        }
+
+        @Specialization(guards = {
+                "isLazyIntRope(rope)",
+                "doesNotRequireBaseConversion(fixBase)"
+        })
+        protected Object stringToInumIntRope(Rope rope, int fixBase, boolean strict, boolean raiseOnError) {
+            return ((LazyIntRope) rope).getValue();
+        }
 
         @Specialization
```

Comment:
We need a negated guard here, so this specialization is never taken fro a LazyIntRope with base=0||10